src/game_battlealgorithm.cpp

```diff
@@ -1312,7 +1349,7 @@ void Game_BattleAlgorithm::Skill::Apply() {
 	AlgorithmBase::Apply();
 
 	for (auto& sa: shift_attributes) {
-		GetTarget()->ShiftAttributeRate(sa, healing ? 1 : -1);
+		GetTarget()->ShiftAttributeRate(sa, (healing ^ negative_effect) ? 1 : -1);
```

Comment:
This is one part where the code gets super confusing, when I first read this I thought you were reversing the attribute shift but you're not because both of these flags get set at the same time.

This inline xor everywhere is super ugly. Can you please wrap this in an inline helper function with a useful name so we can know whats going on.

Btw for this stuff you can push new commits instead of rebasing if thats easier.

Reply:
Moved the XOR-Operations into own functions.